client/alloc_runner.go

```diff
@@ -116,9 +116,9 @@ type allocRunnerState struct {
 	} `json:"Context,omitempty"`
 }
 
-// allocRunnerImmutableState is state that only has to be written once as it
-// doesn't change over the life-cycle of the alloc_runner.
-type allocRunnerImmutableState struct {
+// allocRunnerAllocState is state that only has to be written when the alloc
+// changes.
+type allocRunnerAllocState struct {
```

Comment:
Lets just pull out the version and keep the old immutable object with it in it. Will save us pain later if we do want to add things there and version would endup in two spots